plugins/package-managers/gradle-inspector/src/main/kotlin/GradleInspector.kt

```diff
@@ -371,7 +376,11 @@ private fun createRemoteArtifact(
     val checksum = okHttpClient.downloadText("$artifactUrl.$algorithm")
         .getOrElse { return RemoteArtifact.EMPTY }
 
-    return RemoteArtifact(artifactUrl, parseChecksum(checksum, algorithm))
+    // Ignore file with zero byte size, because it cannot be a valid archive.
+    val hash = parseChecksum(checksum, algorithm).takeIf { it.value != ZERO_BYTES_FILE_SHA1 }
```

Comment:
Nit: Do better document the expected default, consider `takeUnless { it == ZERO_BYTES_FILE_SHA1 }`.